src/syft/core/store/store_memory.py

```diff
@@ -59,19 +60,17 @@ def keys(self) -> KeysView[UID]:
     def values(self) -> ValuesView[AbstractStorableObject]:
         return self._objects.values()
 
-    @syft_decorator(typechecking=True)
-    def store(self, obj: AbstractStorableObject) -> None:
```

Comment:
MAYBE: The ```MemoryStore``` class implements the ```ObjectStore``` which has the ```store``` method (which is ```NotImplemented````).
I am keen to keep the method ```store``` as it is (and also add a ```delete``` method maybe?). We use is as a dict because it is basically a dictionary wrapped inside a class. 

If we were to add more storage types we should adhere to an interface:
* In memory DB (Redis/Memchached)
* File System Based (SQL/NO-SQL DBs)

I think the question is:
Should we adhere to store[object.id_at_location] = object to save objects and del store[object.id_at_location] to remove them, or should we have ```store``` and ```delete``` as methods (and inside those methods we do the "internal" stuff depending on the implementation).

I like the second approach because using it like ```store[object.id_at_location]``` and ```del[object.id_at_location]```` alludes to the store being a dictionary - and might not necessarily be one.

If we go ahead with using the ```dictionary``` methods to delete/add an object then I think we should remove the ```store``` method from the ```ObjectStore```

Reply:
Totally agree, I just wanted to make sure something wasn't happening due to code using both different code paths to mutate the internal dict and the use of UID as the dict key versus passing in an object with a UID set on the StorableObject You are right, the interface should be abstracted from the concept of a dict. I think the naming of "store" as the verb could be changed to "save" so that the code doesn't use: store.store(obj). Also the client store interface exposes the objects as a list which kind of requires indexing so perhaps we could make a distinction there. I have created a ticket, let me know if you think the task requirements need modification: https://github.com/OpenMined/PySyft/issues/4756